flytekit/sdk/tasks.py

```diff
@@ -1000,3 +1001,156 @@ def wrapper(fn):
         return wrapper(_task_function)
     else:
         return wrapper
+
+
+def pytorch_task(
+        _task_function=None,
+        cache_version='',
+        retries=0,
+        interruptible=True,
+        deprecated='',
+        cache=False,
+        timeout=None,
+        workers_count=1,
+        instance_storage_request="",
```

Comment:
Any reason to call it "instance_" ? is it to signify this request is per worker instead of for the entire task?

Reply:
Yes, you've got it right: it's just to stress an accent that it's for requesting resources per instance (both for worker(s) and master)

Reply:
maybe for the user we could provide an interface that says "master_cpu" "worker_cpu". But for now internally map it. Eventually I think we want to use - https://github.com/pytorch/elastic/tree/master/kubernetes, which will not require the master/worker model

Reply:
actually thinking more about it, lets replace "instance" -> "node"? and that should be it. WE could move to elastic later on?

Reply:
Huh, I see that in related docs (at least [here](https://pytorch.org/tutorials/intermediate/dist_tuto.html) and [here](https://www.kubeflow.org/docs/components/training/pytorch/)) processes in distributed job are referred as `replicas`. Hence, may be let's go with `per_replica_*`? Any objections?

Reply:
Do we create master/workers? I don't see that, I only see "workers" created by the operator.
Either way, I think the naming between `X_count` and `per_X_cpu_request` should be consistent... I'm fine with worker, so `workers_count` and `per_worker_cpu_request`... etc. The operator docs are all over the place and inconsistent. Look at this line `You can scale the number of nodes by adjusting .spec.replicaSpecs[Worker].replicas and applying the change.` they have 3 names for pretty much the same thing in one sentence... 
I'm fine with replicas too... as long as we change the count to be `replicas_count` and `per_replica_cpu_request` ... etc.

Reply:
> I only see "workers" created by the operator
Are you referring to extra argument `workers_count`? There is no respective argument for master since there could be the only one. But eventually 1 master and N workers gonna be spawned.

Reply:
i think Haytham means that we should change all the `instance_blah_limit` to `per_worker_blah_limit`

Reply:
since these are per worker right?

Reply:
@EngHabu I think what @igorvalko is trying to do is use same cpu/mem for both master and worker. Initially he had all resources as part of the cRD and i said you could re-purpose our "container" resources for one of them.
So Igor made them the same. 
Now maybe the way to think about this this, the container in "flyte" for a distributed job refers to the master or worker?

Reply:
I would like to avoid the Spark issue we had where people would assign huge resources to the "driver" pod the same way they assign them to the executor pods but they end up barely using any on the driver pod... by essentially forcing users to use the same for both, are we getting into the same situation?

Reply:
I have no hands-on experience with pytorch, but it looks like master in pytorch does the same kind of job as workers with some extra communication arrangement work. In spark different approach is used: driver in most cases is a lightweight coordinator and all the heavy stuff happens on executors.
Hence I assume that's uncommon thing to request uneven resources for master and workers. Input from anyone having experience with distributed pytorch would be very welcome.

Reply:
From what I read, I think there are two possible paradigms: (1) a master node does what a worker does + the coordination and the communication, (2) a master node does only the reduction. But I think either of these requires the master node to have the same/similar resource requirement as workers.

Reply:
(I am assuming we are using `torch.nn.parallel.DistributedDataParallel()` which builds on top of `torch.distributed` package, which should supposedly use the NCCL backend according to https://pytorch.org/docs/stable/distributed.html#which-backend-to-use)